src/helpers/channels/create_stage_instance.ts

```diff
@@ -0,0 +1,39 @@
+import { validateLength } from "../../util/validate_length.ts";
+import { Errors } from "../../types/discordeno/errors.ts";
+import { rest } from "../../rest/rest.ts";
+import { endpoints } from "../../util/constants.ts";
+import { StageInstance } from "../../types/channels/stage_instance.ts";
+import { cacheHandlers } from "../../cache.ts";
+import { ChannelTypes } from "../../types/channels/channel_types.ts";
+import { requireBotChannelPermissions } from "../../util/permissions.ts";
+
+/** Creates a new Stage instance associated to a Stage channel. Requires the user to be a moderator of the Stage channel. */
+export async function createStageInstance(channelId: bigint, topic: string) {
+  const channel = await cacheHandlers.get("channels", channelId);
+  if (!channel) throw new Error(Errors.CHANNEL_NOT_FOUND);
```

Comment:
Do not throw an error if the channel is not cached instead use `if (channel) // do checks`